galley/pkg/runtime/state.go

```diff
@@ -154,3 +154,21 @@ func (s *State) envelopeResource(event resource.Event) (*mcp.Envelope, bool) {
 
 	return entry, true
 }
+
+// String implements fmt.Stringer
+func (s *State) String() string {
+	var b bytes.Buffer
+
+	var messages []string
+	for k := range s.entries {
+		messages = append(messages, k.String())
+	}
+	sort.Strings(messages)
+
+	fmt.Fprintf(&b, "[State @%v]\n", s.versionCounter)
```

Comment:
nit: move `var b bytes.Buffer` closer to its usage,e.g.

```golang
var b bytes.Buffer
fmt.Fprintf(&b, "[State @%v]\n", s.versionCounter)
fmt.Fprintf(&b, "%v", s.buildSnapshot())
```

Also, I don't see `messages` used anywhere.

Reply:
Nice catch. Thanks. It was left-over code from the previous iteration which directly encoded state output in String(). I later switched to simply creating the snapshot.String(), and forgot to remove this.